src/ast.ts

```diff
@@ -88,7 +88,7 @@ interface Caption extends HasAttributes {
 
 interface Table extends HasAttributes {
   tag: "table";
-  children: (Row | Caption)[];
+  children: [Caption, ...Row[]];
```

Comment:
I wonder if we should make this `[...Row[], Caption]` instead? Caption is trailing syntactically, and it seems better to not flip order of children in the AST without really good reason?

Reply:
My only thought there is that with the current order, it's easy to check children[0] and know it's the caption, then iterate through the rest starting with 1.  With the other approach...well I suppose you can check (length - 1) and then iterate from 0 .. length - 2, so maybe it doesn't matter much.  (Haskell instincts: first element of list is easier to access than last element.)